lib/events/events.go

```diff
@@ -351,3 +353,54 @@ func Error(err error) *string {
 	str := err.Error()
 	return &str
 }
+
+// Wait for the given event to become available, or for a timeout, whichever is sooner.
+// Returns true if the event became available, or false if we timed out.
+func (s *bufferedSubscription) waitForNewEventWithTimeout(id int, timeout time.Duration) bool {
+	// Checks whether 'id' is available. If it is, returns nil. If it isn't, returns newEventNotification (constructing it if necessary)
+	checkIDAndConstructNotificationIfNecessary := func() <-chan struct{} {
```

Comment:
Why do you need a list of waiting channels? Can't you share one channel between all waiters, as I've done?